spark/interpreter/src/main/java/org/apache/zeppelin/spark/ZeppelinR.java

```diff
@@ -391,4 +397,24 @@ public void onProcessFailed(ExecuteException e) {
     logger.error(e.getMessage(), e);
     rScriptRunning = false;
   }
+
+
+  public static class SparkRInterpreterOutputStream extends InterpreterOutputStream {
+
+    private SparkRInterpreter sparkRInterpreter;
+
+    public SparkRInterpreterOutputStream(Logger logger, SparkRInterpreter sparkRInterpreter) {
+      super(logger);
+      this.sparkRInterpreter = sparkRInterpreter;
+    }
+
+    @Override
+    protected void processLine(String s, int i) {
+      super.processLine(s, i);
+      if (s.contains("Java SparkR backend might have failed") // spark 2.x
+          || s.contains("Execution halted")) { // spark 1.x
```

Comment:
`Execution halted` might not be specific enough - I think it could happen in a number of situations

Reply:
Unfortunately, this is the only useful information that I can use for checking whether sparkr jvm backend is dead.  Here's the message I can get
```
13:41:01,157 DEBUG org.apache.zeppelin.spark.ZeppelinR:81 - Interpreter output:Error in if (returnStatus != 0) { : argument is of length zero
13:41:01,157 DEBUG org.apache.zeppelin.spark.ZeppelinR:81 - Interpreter output:Calls: <Anonymous> -> invokeJava
13:41:01,157 DEBUG org.apache.zeppelin.spark.ZeppelinR:81 - Interpreter output:Execution halted
```